scipy/interpolate/tests/test_bsplines.py

```diff
@@ -3533,6 +3533,18 @@ def test_s_too_small(self):
             xp_assert_close(np.r_[spl.c, [0]*(spl.k+1)],
                             tck[1], atol=5e-13)
 
+        x = np.asarray([20.00, 153.81, 175.57, 202.47, 237.11,
+             253.61, 258.56, 273.40, 284.54, 293.61,
+             298.56, 301.86, 305.57, 307.22, 308.45,
+             310.10, 310.10, 310.50], dtype=np.float64)
+        y = np.asarray([53.00, 49.50, 48.60, 46.80, 43.20,
+             40.32, 39.60, 36.00, 32.40, 28.80,
+             25.20, 21.60, 18.00, 14.40, 10.80,
+             7.20, 3.60, 0.0], dtype=np.float64)
+        w = np.asarray([1.38723] * y.shape[0], dtype=np.float64)
+        with assert_raises(RuntimeError):
+            make_splrep(x, y, w=w, k=2, s=12)
```

Comment:
Could you please also add an assertion that `np.isnan(splrep(x, y, w=w, k=k, s=s)[1])`, if only to simplify "what the ... " headscratching next time we'll need to `git blame` it, years from now.

Reply:
Hm interesting. So this might be the reason behind https://github.com/scipy/scipy/issues/22704#issuecomment-2873750711.
Is 
```
tck = splrep(x, y, w=w, k=k, s=s)
splev(x, tck)
```
reasonable or is garbage on your system?

Reply:
```zsh
[52.91315852 50.14097155 48.59291111 46.25490503 42.55321297 40.51648816
 39.7754699  36.31304728 32.49105859 28.60476531 25.27998696 21.93667047
 17.1015815  14.58516652 11.64192351  4.60491033  4.60491033  1.10886194]
```

Reasonable and somewhat close to `y`.